File: examples/ios-objc/RudderTyperExample/RudderConfig/RudderConfig.m
//
//  RudderConfig.m
//  RudderTyperExample
//
//  Created by Satheesh Kannan on 08/08/24.
//

#import "RudderConfig.h"

#pragma mark - RudderConfig
@implementation RudderConfig
/**
 * This funtion will initialize this model with all properties.
 */
- (instancetype)initWithDictionary:(NSDictionary *)dictionary {
    self = [super init];
    if (self) {
        _writeKey = dictionary[@"WRITE_KEY"] ?: @"";
        _dataPlaneUrl = dictionary[@"DATA_PLANE_URL"] ?: @"";
        _controlPlaneUrl = dictionary[@"CONTROL_PLANE_URL"] ?: @"";
    }
    return self;
}

@end

File: tests/e2e/ios-swift/RudderTyperSwiftExample/RudderConfig/RudderConfig.swift
//
//  RudderConfig.swift
//  RudderTyperSwiftExample
//
//  Created by Satheesh Kannan on 12/08/24.
//

import UIKit

struct RudderConfig: Codable {
    let writeKey: String
    let dataPlaneUrl: String
    let controlPlaneUrl: String
    
    enum CodingKeys: String, CodingKey {
        case writeKey = "WRITE_KEY"
        case dataPlaneUrl = "DATA_PLANE_URL"
        case controlPlaneUrl = "CONTROL_PLANE_URL"
    }
}
